### PR TITLE
QKeras updates

### PR DESCRIPTION
**Pull request by thesps**
New specific things:

- Add an Optimizer pass to set the rounding / saturation mode for the layer output. This is necessary to get decent with performance with the low precision quantized_relu layers. The pass is effectively disabled by default, since it may not be necessary for quantized_relu layers with higher precision.
- New softmax implementation. The softmax implementation was quite error prone, and contained several hard-coded values, so I have rewritten it. The new version also has an extra configurability: each LUT precision can be set independently. Although even without this configurability, the new version is more reliable.

### CODE REVIEW

**Comment by thesps** [comment]
The new `softmax` implementation needs to be propagated to the other converters. Since it is represented by a new layer class, we need to make sure the converter instantiates the right layer (the new `Softmax`, rather than `Activation).

**Comment by thesps** [comment]
I added the new softmax layer (with its two tables) to the other converters. I tested the onnx and TF models (successfully) but couldn't try to Pytorch one yet due to (I think) some issue with that model being from a much older Pytorch version.
